[TECH] Mettre à jour des linters sur Pix App (PIX-1634).

sbedeau opened the pull request:
## :unicorn: Problème
Les linters ne sont pas à jour sur Pix App. 

Une mise en conformité est également nécessaire : 
<img width="1676" alt="Screenshot 2020-11-06 at 17 32 01" src="https://user-images.githubusercontent.com/46371288/98395421-84ade100-205c-11eb-9983-e8783db145ab.png">

## :robot: Solution
Mise à jour et mise en conformité de la règle [no-controller-access-in-routes](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-controller-access-in-routes.md)

## :rainbow: Remarques
RAS

## :100: Pour tester
- [x] Vérifier `campaign-landing-page`
- [x] Vérifier `fill-in-campaign-code`
- [x] Vérifier `fill-in-certificate-verification-code`

pix-service commented:
I'm deploying this PR to these urls:

- App: https://app-pr2120.review.pix.fr
- Orga: https://orga-pr2120.review.pix.fr
- Certif: https://certif-pr2120.review.pix.fr
- Admin: https://admin-pr2120.review.pix.fr
- API: https://api-pr2120.review.pix.fr/api/

 Please check it out!

Commit:
Lint fix - prefer find method from @ember/test-helpers

Commit:
Apply ember/no-controller-access-in-routes for campaign-landing-page

Commit:
Apply ember/no-controller-access-in-routes for fill-in-campaign-code

Commit:
Apply ember/no-controller-access-in-routes for fill-in-certificate-verification-code

Commit:
Bump eslint

Commit:
Bump eslint-plugin-ember

MelanieMEB commented on mon-pix/app/mixins/progression-tracker.js:
```diff
@@ -17,9 +18,9 @@ export default Mixin.create({
     this._super(...arguments);
-    this.stepsAhead = this.steps.map(
+    set(this, 'stepsAhead', this.steps.map(
       (step) => EmberObject.create({ name: step, status: STEPS.TO_COMPLETE }),
-    );
+    ));
```
**question :** Pourquoi ce changement ? Je pensais qu'on utilisait plus les set/get ?

sbedeau replied:
Bonne question, ça a piqué ma curiosité. J'avais lancé un autofix du linter au moment du bump et j'ai donc splitté le commit en intention. 

L'intention du fix dans ce fichier était la suivante : 
> error  Use `set(this, 'propertyName', 'value')` instead of assignment for untracked properties that are used as computed property dependencies (or convert to using tracked properties)  ember/no-assignment-of-untracked-properties-used-in-tracking-contexts

Je vais voir pour passer en tracked afin de supprimer le set sinon je laisse comme ça

jbuget replied:
Après discussion avec @sbedeau , on décide d'avancer en laissant tel quel pour le moment, en espérant que d'ici 6 mois (18 si on est honnêtes avec nous-mêmes) les mixins aient disparues des Pix fronts.

Commit:
Bump eslint-plugin packages

Commit:
Remove use of get method from ember/object

Commit:
Lint fix - Remove assignment use in mixin

error Use set(this, 'propertyName', 'value') instead of assignment for untracked properties that are used as computed property dependencies (or convert to using tracked properties) ember/no-assignment-of-untracked-properties-used-in-tracking-contexts

We chose to keep the use of the set method because it is in a mixin.
Mixin are supposed to dispear in time with glimmer; they are tolerated for now.

Commit:
Bump eslint

Commit:
Bump stylelint

Commit:
Bump ember-template-lint

Commit:
Bump eslint package
